pkg/scheduler/util/utils.go

```diff
@@ -95,5 +95,12 @@ func (l *SortableList) Sort() {
 // the second one. It takes arguments of the type "interface{}" to be used with
 // SortableList, but expects those arguments to be *v1.Pod.
 func HigherPriorityPod(pod1, pod2 interface{}) bool {
+	if GetPodPriority(pod1.(*v1.Pod)) == GetPodPriority(pod2.(*v1.Pod)) {
```

Comment:
This change is risky. When we have many unschedulable pods with the same priority, the pods which are created earlier, will go to the head of the queue and could block other pods from getting scheduled.

Reply:
Hey sorry for the late reply as I was trying to understand the blocking scenario. I created the following scenario for testing this. 
*2 node cluster with CPU Allocatable as 1 on each*
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: test-pod-1
spec:
  containers:
  - name: nginx
    image: nginx
    resources:
      requests:
        cpu: 0.5
---
apiVersion: v1
kind: Pod
metadata:
  name: test-pod-2
spec:
  containers:
  - name: nginx
    image: nginx
    resources:
      requests:
        cpu: 0.5
---
apiVersion: v1
kind: Pod
metadata:
  name: test-pod-3
spec:
  containers:
  - name: nginx
    image: nginx
    resources:
      requests:
        cpu: 0.5
---
apiVersion: v1
kind: Pod
metadata:
  name: test-pod-4
spec:
  containers:
  - name: nginx
    image: nginx
    resources:
      requests:
        cpu: 0.5
---
apiVersion: v1
kind: Pod
metadata:
  name: test-pod-5
spec:
  containers:
  - name: nginx
    image: nginx
    resources:
      requests:
        cpu: 0.5
---
apiVersion: v1
kind: Pod
metadata:
  name: test-pod-6
spec:
  containers:
  - name: nginx
    image: nginx
    resources:
      requests:
        cpu: 0.5
---
apiVersion: v1
kind: Pod
metadata:
  name: test-pod-7
spec:
  containers:
  - name: nginx
    image: nginx
    resources:
      requests:
        cpu: 1
---
apiVersion: v1
kind: Pod
metadata:
  name: test-pod-8
spec:
  containers:
  - name: nginx
    image: nginx
    resources:
      requests:
        cpu: 0.5
---
apiVersion: v1
kind: Pod
metadata:
  name: test-pod-9
spec:
  containers:
  - name: nginx
    image: nginx
    resources:
      requests:
        cpu: 1
``` 
First `test-pod-1`, `test-pod-2`, `test-pod-3`, `test-pod-4` get scheduled.
deleting `test-pod-1` triggers `test-pod-5` to get schedule 
deleting `test-pod-2` triggers `test-pod-6` to get schedule
deleting `test-pod-3` triggers `test-pod-8` to get schedule
`test-pod-7` does not have available resources so tries and fails and does not block `test-pod-8` from getting schedule after resources become available between `test-pod-7` and `test-pod-9`... `test-pod-7` will get schedule.

Reply:
I explained this in our SIG Scheduling meeting today, but for the record, the issue happens in larger clusters where node updates are frequent. Each node sends a node update every ten seconds. In clusters with thousands of nodes, there are hundreds of node updates per second. These node updates cause the scheduler to move all the unschedulable pods from the unschedulable queue to the active queue and retry scheduling them. So, a pod with the same priority that is created ahead of other similar pods goes to the head of the queue very frequently and blocks and slows down scheduling of other pods.

Reply:
@sudeshsh - Not sure, how far you have gone along with your implementation of adding a flag to scheduler for this change, could you use an extender for preemption and have this logic there? I'd recommend doing that as the change you're making may end up as a throw-away work. You can track it from here - https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/core/extender_test.go#L197

Reply:
We get similar requirements when running [KubeGene](http://github.com/kubegene/kubegene) with default scheduler. How about ratelimites if pod failed to scheduled (Priority > Ratelimites > CreationTime)?

Reply:
@k82cn - You mean something like backoffs - https://github.com/kubernetes/kubernetes/pull/57057?

Reply:
yes, similar :)

Reply:
@ravisantoshgudimetla Yes waiting for the backoff PR to go in rather than implementing the short-term flag solution.